src/Platform/Sdl/SdlWindow.cpp

```diff
@@ -73,23 +73,44 @@ bool SdlWindow::IsVisible() const {
     return false;
 }
 
-void SdlWindow::SetFullscreen(bool fullscreen) {
-    if (SDL_SetWindowFullscreen(window_, fullscreen ? SDL_WINDOW_FULLSCREEN : 0) != 0)
+void SdlWindow::SetMode(bool fullscreen, bool borderless) {
```

Comment:
The changes go against the API design principles outlined in Platform.h.

What's the reasoning behind these changes? Maybe I can offer a better way to do it.

Like:

enum class WindowMode {
    Fullscreen,
    Windowed
}

Fullscreen switches into SDL_WINDOW_FULLSCREEN_DESKTOP / SDL_WINDOW_FULLSCREEN depending on the window size.

Windowed always implies !SDL_WINDOW_BORDERLESS, so that we don't support borderless non-fullscreen windows.

Reply:
Or alternatively can explicitly provide different enums for different fullscreen modes

enum class WindowMode {
    FullscreenDesktop,
    FullscreenNative,
    Windowed
}